### PR TITLE
Intel quicky 1 to implement 1

### LINKED ISSUE #40: There should be a way to reset the reducer
**Describe the feature or change you'd like**
Currently it is not possible to reset the reducer without resetting the whole intel. This should be changed. It should be possible to reinit the reducer separately.

### LINKED ISSUE #41: There should be a gitlab implementation
**Describe the feature or change you'd like**
Currently the gitlab util is not implemented in the intel. Therefore it cannot be used. It should be implemented in order to support source objects with gitlab as platform type.

### LINKED ISSUE #42: The squeezer should support excluding keys
**Describe the feature or change you'd like**
Currently the squeezer only supports key filtering. It does not support excluding keys. This functionallity should be added.

### LINKED ISSUE #43: There should be refined exclude options in the tsconfig
**Describe the feature or change you'd like**
Currently the folder "lib" is not included in the exclude options. This is required in order to generate the module fluently.

**Additional context**
image

### LINKED ISSUE #44: There should be a support for resolving json modules
**Describe the bug**
Currently it is not possible to build a package because OAuth uses a JSON config file which leads to an error because resolving json modules is not activated.

**To Reproduce**
1. type "npx tsc"

**Expected behavior**
It should build without errors.
**Additional context**
image

### LINKED ISSUE #46: There should be repository owner information without losing organization information
**Describe the feature or change you'd like**
A repository is owned by the organization in which it is located. Therefore, the organization is rendered to the extent that it only contains the data that match a member. Then this rendered organization is saved as the owner.

It is important that the original data of the organization remain.

This does not happen. All data that do not match the member data are deleted during the rendering process. Therefore information like the members of an organisation is lost.

To solve this, a deep copy of a organisation must be made. This is then rendered and used as the owner.

Thus, the original data remain. 🎉 

**Additional context**
Ref: https://www.npmjs.com/package/lodash.clonedeep

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no way to reset the reducer. #40 
- Currently there is no gitlab implementation. #41
- Currently the squeezer does not support key excluding. #42
- Currently a folder is missing in the exclude options. #43 
- Currently there is no support for json modules. #44
- Currently the repository owner eliminates important organisation data.  #46

**What is the new behavior (if this is a feature change)?**
- Now there is a way to reset the reducer. Closes #40 
- Now gitlab is implemented and can be used. Closes #41 
- Now the squeezer does support key excluding. Closes #42 
- Now the folder "lib" is added to the exclude options. Closes  #43 
- Now resolving json modules is supported. Closes #44
- Now the repository owner does not influence organisational data. Closes #46

**Other information**:
-  Ref: 🐍

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,13 +14,15 @@
 //#PACKAGE snek-client
 //## npm install snek-client
 // Contains the clients for API calls to SNEK and GitHub.
```
Comment (schettn) [inline src/index.ts:16]: Gitlab is missing.

File: src/index.ts [hunk 2]
```diff
@@ -14,13 +14,15 @@
...
-import { SnekClient, GithubClient } from "snek-client";
+import { SnekClient, GithubClient, GitlabClient } from "snek-client";
 //> Reducer
 // Contains the reducer and database models
 import { Reducer } from "./reducer";
 //> Utils
 // Contains the github utils
 import * as github from "./utils/github/index";
+// Contains the github utils
```
Comment (schettn) [inline src/index.ts:24]: Rename to: Contains the github util

File: src/index.ts [hunk 3]
```diff
@@ -14,13 +14,15 @@
 //#PACKAGE snek-client
 //## npm install snek-client
 // Contains the clients for API calls to SNEK and GitHub.
-import { SnekClient, GithubClient } from "snek-client";
+import { SnekClient, GithubClient, GitlabClient } from "snek-client";
 //> Reducer
 // Contains the reducer and database models
 import { Reducer } from "./reducer";
 //> Utils
 // Contains the github utils
```
Comment (schettn) [inline src/index.ts:22]: Rename to: Contains the github util

File: src/index.ts [hunk 4]
```diff
@@ -13,14 +13,16 @@
 //> snek-client
 //#PACKAGE snek-client
 //## npm install snek-client
-// Contains the clients for API calls to SNEK and GitHub.
-import { SnekClient, GithubClient } from "snek-client";
+// Contains the clients for API calls to SNEK, Github and Gitlab.
+import { SnekClient, GithubClient, GitlabClient } from "snek-client";
 //> Reducer
```
Comment (pinterid) [inline src/index.ts:18]: Add newline.

File: src/index.ts [hunk 5]
```diff
@@ -78,25 +82,33 @@ interface ISource {
   authorization: string;
 }
 
-/** @interface DataUser defines the structure of the reponse of the github client. */
+/**
+ *  @interface DataUser defines the structure of the response of
```
Comment (pinterid) [inline src/index.ts:86]: from the github client?

File: src/reducer/helper/profile.ts [hunk 6]
```diff
@@ -30,9 +33,9 @@ interface IPlatform extends models.Platform {
  */
 interface IRepository extends models.Repository {
   /**
-   * Owner: A member object.
+   * Owner: A organization object.
    */
-  owner: models.Member;
+  owner: models.Organization;
```
Comment (kleberbaum) [inline src/reducer/helper/profile.ts:38]: ?

**Pull request merged**